model/Service/ConcurringSessionService.php

```diff
@@ -67,6 +77,21 @@ public function __construct(
         $this->currentSession = $currentSession ?? PHPSession::singleton();
     }
 
+    public function pauseActiveDeliveryExecution($activeExecution): void
+    {
+        if ($activeExecution instanceof DeliveryExecution) {
+            $this->getConcurringSessionService()->pauseConcurrentSessions($activeExecution);
+
+            if ($activeExecution->getState()->getUri() === DeliveryExecution::STATE_PAUSED) {
+                $this->getConcurringSessionService()->adjustTimers($activeExecution);
+            }
+
+            $this->getConcurringSessionService()->clearConcurringSession($activeExecution);
+        }
+
+        $this->resetDeliveryExecutionState($activeExecution);
```

Comment:
`resetDeliveryExecutionState` does nothing if `$activeExecution` is `null`, moving that method call inside the conditional body may be a bit more clear.

```suggestion
        if ($activeExecution instanceof DeliveryExecution) {
            $this->getConcurringSessionService()->pauseConcurrentSessions($activeExecution);

            if ($activeExecution->getState()->getUri() === DeliveryExecution::STATE_PAUSED) {
                $this->getConcurringSessionService()->adjustTimers($activeExecution);
            }

            $this->getConcurringSessionService()->clearConcurringSession($activeExecution);
            $this->resetDeliveryExecutionState($activeExecution);
        }
```